cpp/test/spatial/ball_cover.cu

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <raft/cudart_utils.h>
+#include <raft/linalg/distance_type.h>
+#include <iostream>
+#include <raft/spatial/knn/ball_cover.hpp>
+#include <raft/spatial/knn/detail/haversine_distance.cuh>
+#include <raft/spatial/knn/detail/knn_brute_force_faiss.cuh>
+#include <rmm/device_uvector.hpp>
+#include <vector>
+#include "../test_utils.h"
+#include "spatial_data.h"
+
+#include <thrust/transform.h>
+#include <rmm/exec_policy.hpp>
+
+namespace raft {
+namespace spatial {
+namespace knn {
+
+using namespace std;
```

Comment:
It's probably best to avoid this even in tests. We only do it in a couple other test files, and I'd be in favor of not adding any more.